Cython/Tests/TestTestUtils.py

```diff
@@ -72,3 +74,56 @@ def test_write_newer_file_fresh(self):
         self.assertFalse(os.path.exists(file_path))
         write_newer_file(file_path, file_path, "xyz")
         self.assertTrue(os.path.isfile(file_path))
+
+    ################################# Sandbox ##################################
+
+    def _sandbox_set_up(self):
+        Utils._function_caches = self._function_caches
+
+    def sandboxed(self):
+        # setup value
+        self.assertEqual(Utils._function_caches, self._function_caches)
+
+        Utils._function_caches = [
+            "What is the airspeed velocity of an unladen swallow?",
+            "I don't know that"]
+
+    def _test_sandbox_for_function_caches(self, add_setup_func, asserted):
+        self.assertFalse(hasattr(self, "set_up_test_in_sandbox"))
+        if add_setup_func:
+            self.set_up_test_in_sandbox = self._sandbox_set_up
+
+        caches_before = Utils._function_caches
+        caches_contents = Utils._function_caches[:]
+        caches_copy = deepcopy(Utils._function_caches)
+
+        sandboxed = sandbox_for_function_caches(asserted)(self.sandboxed)
+
+        self.assertIs(caches_before, Utils._function_caches)
+        self.assertEqual(caches_copy, Utils._function_caches)
```

Comment:
It is assumed that the sandbox will not change the values of the caches, for confirmation I use `deepcopy`, some tests do not pass line `self.assertEqual(caches_copy, Utils._function_caches)` (more precisely, only versions 2.7 and 3.6)
Is it worth exploring this in more detail, or just write a comment about the expected behavior and stop using `deepcopy`?